partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/CastTable.kt

```diff
@@ -0,0 +1,701 @@
+        Array(PType.Kind.values().size) { targetKind ->
+            null
+        }
+    }
+
+    init {
+        registerBool()
+        registerTinyInt()
+        registerSmallInt()
+        registerInt()
+        registerBigInt()
+        registerIntArbitrary()
+        registerDecimal()
+        registerDecimalArbitrary()
+        registerReal()
+        registerDoublePrecision()
+        registerStruct()
+        registerString()
+        registerSymbol()
+        registerBag()
+        registerList()
+        registerSexp()
+    }
+
+    /**
+     * CAST(<bool> AS <type>)
+     * TODO: CHAR, VARCHAR, SYMBOL
+     */
+    private fun registerBool() {
+        register(PType.Kind.BOOL, PType.Kind.BOOL) { x, _ -> x }
+        register(PType.Kind.BOOL, PType.Kind.TINYINT) { x, _ -> Datum.tinyInt(if (x.boolean) 1 else 0) }
+        register(PType.Kind.BOOL, PType.Kind.SMALLINT) { x, _ -> Datum.smallInt(if (x.boolean) 1 else 0) }
+        register(PType.Kind.BOOL, PType.Kind.INT) { x, _ -> Datum.int32Value(if (x.boolean) 1 else 0) }
+        register(PType.Kind.BOOL, PType.Kind.BIGINT) { x, _ -> Datum.int64Value(if (x.boolean) 1 else 0) }
+        register(
+            PType.Kind.BOOL,
+            PType.Kind.INT_ARBITRARY
+        ) { x, _ -> Datum.intArbitrary(if (x.boolean) BigInteger.ONE else BigInteger.ZERO) }
+        register(
+            PType.Kind.BOOL,
+            PType.Kind.DECIMAL
+        ) { x, t -> Datum.decimal(if (x.boolean) BigDecimal.ONE else BigDecimal.ZERO, t.precision, t.scale) }
+        register(
+            PType.Kind.BOOL,
+            PType.Kind.DECIMAL_ARBITRARY
+        ) { x, _ -> Datum.decimalArbitrary(if (x.boolean) BigDecimal.ONE else BigDecimal.ZERO) }
+        register(PType.Kind.BOOL, PType.Kind.REAL) { x, _ -> Datum.real(if (x.boolean) 1F else 0F) }
+        register(
+            PType.Kind.BOOL,
+            PType.Kind.DOUBLE_PRECISION
+        ) { x, _ -> Datum.doublePrecision(if (x.boolean) 1.0 else 0.0) }
+        register(PType.Kind.BOOL, PType.Kind.STRING) { x, _ -> Datum.stringValue(if (x.boolean) "true" else "false") }
+        register(PType.Kind.BOOL, PType.Kind.SYMBOL) { x, _ -> Datum.stringValue(if (x.boolean) "true" else "false") }
+    }
+
+    /**
+     * CAST(<tinyint> AS <type>)
+     * TODO: CHAR, VARCHAR, SYMBOL
+     */
+    private fun registerTinyInt() {
+        register(PType.Kind.TINYINT, PType.Kind.BOOL) { x, _ -> Datum.boolValue(x.byte.toInt() != 0) }
+        register(PType.Kind.TINYINT, PType.Kind.TINYINT) { x, _ -> x }
+        register(PType.Kind.TINYINT, PType.Kind.SMALLINT) { x, _ -> Datum.smallInt(x.byte.toShort()) }
+        register(PType.Kind.TINYINT, PType.Kind.INT) { x, _ -> Datum.int32Value(x.byte.toInt()) }
+        register(PType.Kind.TINYINT, PType.Kind.BIGINT) { x, _ -> Datum.int64Value(x.byte.toLong()) }
+        register(PType.Kind.TINYINT, PType.Kind.INT_ARBITRARY) { x, _ ->
```

Comment:
nit: clean up by importing all of the PType.Kind to reduce a lot of the noise here